src/ConnectionFactory.php

```diff
@@ -120,7 +120,10 @@ public function createConnection(array $params, ?Configuration $config = null, ?
             $driver     = $connection->getDriver();
             /** @psalm-suppress InvalidScalarArgument Bogus error, StaticServerVersionProvider implements Doctrine\DBAL\ServerVersionProvider  */
             $platform = $driver->getDatabasePlatform(
-                ...(class_exists(StaticServerVersionProvider::class) ? [new StaticServerVersionProvider($params['serverVersion'] ?? '')] : []),
+                ...(class_exists(StaticServerVersionProvider::class)
+                    ? [new StaticServerVersionProvider($params['serverVersion'] ?? $params['primary']['serverVersion'] ?? '')]
```

Comment:
need to check this one a bit more: might be a bug

Reply:
This indeed was a bug however not really a big issue. The only thing we need to know further below is if `$platform` is an instance of `AbstractMySQLPlatform` and thats true anyway no matter which mysql version we use.

But this fix is needed to be deprecation free and to use the proper static version then.

This is in line with https://github.com/doctrine/dbal/blob/c70bae9e69e0e543044d5e282a624d9b3cbea6fb/src/Connection.php#L185-L189